eng/pipelines/docindex.yml

```diff
@@ -13,6 +13,18 @@ jobs:
     pool:
       vmImage: windows-2019
     steps:
+      - task: UsePythonVersion@0
```

Comment:
@scbedd orthogonal to Jim's change here but is it worth adding the UsePythonVersion in the template instead of needing to call it explicitly? Or do most pipelines already explicitly setup a python version?

Reply:
The script is designed to be usable on any of the major python versions we leverage. Originally I was hesitant to add to the script itself because we need to restore the path back to what it was prior to running this script. 

@JimSuplizio  did this fail if you didn't UsePythonVersion?

Reply:
Because if that's the case that is unexpected to me and I'd like to resolve that.

Reply:
While not exactly the same I tried running the versioning update scripts without setting this and it failed. https://dev.azure.com/azure-sdk/internal/_build/results?buildId=216201&view=logs&j=a8b3ef96-4a4b-559b-f023-10f869b6ec93&t=b95ceac7-6b57-5697-85af-4d84d95bb884. I suspect it is because it ran on an earlier version of Python. I will verify that buy explicitly setting the python version.